Tests iOS/HomeTests.swift

```diff
@@ -80,22 +87,44 @@ class HomeTests: XCTestCase {
 
         home.sectionHeader("Slate 2").verify()
         home.recommendationCell("Slate 2, Recommendation 1").verify()
+
+        await snowplowMicro.assertBaselineSnowplowExpectation()
+
+        async let slate1Rec1 = snowplowMicro.getFirstEvent(with: "discover.impression", recommendationId: "slate-1-rec-1")
+        async let slate1Rec2 = snowplowMicro.getFirstEvent(with: "discover.impression", recommendationId: "slate-1-rec-2")
+        async let slate2Rec1 = snowplowMicro.getFirstEvent(with: "discover.impression", recommendationId: "slate-2-rec-1")
+        async let slate2Rec2 = snowplowMicro.getFirstEvent(with: "discover.impression", recommendationId: "slate-1-rec-2")
+
+        let recs = await [slate1Rec1, slate1Rec2, slate2Rec1, slate2Rec2]
+        let loadedSlate1Rec1 = recs[0]!
+        let loadedSlate1Rec2 = recs[1]!
+        let loadedSlate2Rec1 = recs[2]!
+        let loadedSlate2Rec2 = recs[3]!
+
+        snowplowMicro.assertRecommendationImpressionHasNecessaryContexts(event: loadedSlate1Rec1, url: "http://localhost:8080/hello")
+        snowplowMicro.assertRecommendationImpressionHasNecessaryContexts(event: loadedSlate1Rec2, url: "https://example.com/item-2")
+        snowplowMicro.assertRecommendationImpressionHasNecessaryContexts(event: loadedSlate1Rec1, url: "https://example.com/item-1")
+        snowplowMicro.assertRecommendationImpressionHasNecessaryContexts(event: loadedSlate1Rec1, url: "https://example.com/item-2")
     }
 
-    func test_navigatingToHomeTab_showsRecentlySavedItems() {
+    @MainActor
+    func test_navigatingToHomeTab_showsRecentlySavedItems() async {
         let home = app.launch().homeView.wait()
         home.savedItemCell("Item 1").wait()
         home.savedItemCell("Item 2").wait()
         home.savedItemCell("Item 1").swipeLeft(velocity: .fast)
         home.savedItemCell("Item 3").swipeLeft(velocity: .fast)
         waitForDisappearance(of: home.savedItemCell("Item 3"))
+        await snowplowMicro.assertBaselineSnowplowExpectation()
```

Comment:
If there a way to add this at the end of every UI test we should. I could not find it.

Reply:
There is tearDown that occurs at the end of every test but typically used for cleanup.

Reply:
I looked into that, but I couldn't tell if it would fail as expected.

Reply:
I added `XCTAssertTrue(false)` in the teardown and it fails the test, so I assume so based on that the snowplow one will work the same.

![image](https://user-images.githubusercontent.com/6743397/220997628-1ce74120-152c-4700-9242-fb2a93ba1e4f.png)